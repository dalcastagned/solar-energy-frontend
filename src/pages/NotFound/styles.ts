import Button from '@mui/material/Button';

import { styled } from '@mui/material/styles';

export const Bot = styled('img')(() => ({
  width: '70rem',
  marginTop: '-20rem',
  marginLeft: '-13rem',
}));

export const Wrapper = styled('div')(() => ({
  display: 'flex',
  width: '100%',
  height: 'calc(100vh - 10.4rem)',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
}));

export const TitleCode = styled('h1')(() => ({
  marginTop: '-13rem',
  marginBottom: '1rem',
}));

export const DetailsCode = styled('p')(() => ({
  marginBottom: '2rem',
}));

export const ReturnButton = styled(Button)(() => ({
  height: '5rem',
  fontSize: '1.6rem',
}));
