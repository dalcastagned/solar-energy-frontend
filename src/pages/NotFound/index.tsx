import * as S from './styles';

const NotFound = (): JSX.Element => (
  <S.Wrapper>
    <S.Bot src="/img/bot.svg" alt="" />
    <S.TitleCode>Oops! Página não encontrada</S.TitleCode>
    <S.DetailsCode>
      A página que você está tentando acessar não existe ou foi movida.
    </S.DetailsCode>
    <S.DetailsCode>Tente voltar para a nossa página inicial</S.DetailsCode>
    <S.ReturnButton variant="contained" href="/">
      Ir para página inicial
    </S.ReturnButton>
  </S.Wrapper>
);

export default NotFound;
