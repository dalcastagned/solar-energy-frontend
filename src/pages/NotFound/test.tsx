import NotFound from '.';
import { renderWithRoute } from '../../utils/tests/helpers';

describe('<NotFound />', () => {
  it('should render the main', () => {
    const { container } = renderWithRoute(<NotFound />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
