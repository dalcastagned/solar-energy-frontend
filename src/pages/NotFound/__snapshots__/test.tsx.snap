// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render the main 1`] = `
<div
  class="css-1clb1mu"
>
  <img
    alt=""
    class="css-1dwhv5t"
    src="/img/bot.svg"
  />
  <h1
    class="css-jftm75"
  >
    Oops! Página não encontrada
  </h1>
  <p
    class="css-8fkoyy"
  >
    A página que você está tentando acessar não existe ou foi movida.
  </p>
  <p
    class="css-8fkoyy"
  >
    Tente voltar para a nossa página inicial
  </p>
  <a
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-126p7k4-MuiButtonBase-root-MuiButton-root"
    href="/"
    tabindex="0"
  >
    Ir para página inicial
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</div>
`;
